### PR TITLE
Apply mantine color palette to cartesian chart labels

### LINKED ISSUE #41949: Cartesian charts labels should inherit color from Mantine theme in React embedding
Cartesian charts' axis labels and legends should inherit the color palette of the user-supplied Mantine theme. The color of the chart labels is currently hard-coded.

CleanShot 2567-04-25 at 22 06 34@2x

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/41949

### Description

Cartesian charts currently get their color from the `color` function from `lib/colors/palette.ts`, but these colors are hard-coded and not overrideable by users. This PR adds the `use-palette` hook that extracts a subset of color palettes from a Mantine theme, and apply the palette into the cartesian chart.

### How to verify

1. Go to the `basic-theming` branch in the demo app
2. Go to the `/admin/internal/kitchen-sink` page
3. Verify that the theme color in `AppProvider` is set to a custom value, e.g. text-dark is hotpink:

```ts
  colors: {
    brand: colorTuple("hotpink"),
    "text-dark": colorTuple("hotpink"),
    "text-light": colorTuple("hotpink"),
  },
```

4. The 3 chart labels should now be in hotpink, instead of black.

### Screenshot

Labels used to use the hard-coded text-dark color, but now it becomes hotpink as defined above.

CleanShot 2567-05-02 at 19 01 08@2x

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/hooks/use-palette.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * Gets a color palette from the current Mantine theme.
+ *
+ * Allows palettes to be overridden by the user, primarily via the React embedding SDK.
+ */
+export function usePalette(): ColorPalette {
+  const theme = useMantineTheme();
+
+  return useMemo(() => {
+    const color = (name: MantineColor) => theme.colors[name]?.[0];
```
Comment (heypoom) [inline frontend/src/metabase/hooks/use-palette.ts:16]: Not sure if it makes sense to use the first color in the array `[0]` here as Mantine has 10 shades. The first color is the lightest color

> Colors are exposed on the theme object as an array of strings, you can access color shade by color name and index (0-9), colors with larger index are darker:

https://mantine.dev/theming/colors/

Comment (WiNloSt) [inline frontend/src/metabase/hooks/use-palette.ts:16]: The idea looks good, but Mantine has a function for this.

```suggestion
    const color = theme.fn.themeColor
```

Or you could just inline that. e.g.
```ts
theme.fn.themeColor("white")
```

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts [hunk 2]
```diff
@@ -49,12 +52,12 @@ export function useModelsAndOption({
 
   const renderingContext: RenderingContext = useMemo(
     () => ({
-      getColor: color,
+      getColor: name => color(name, palette),
```
Comment (WiNloSt) [inline frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts:55]: passing colors we want to override here looks good. I kind of remember we did this when we worked with static viz to make user appearance colors works.

File: frontend/src/metabase/hooks/use-palette.ts (hunk 1 above)
Comment (heypoom) [reply frontend/src/metabase/hooks/use-palette.ts:16]: Didn't knew this, thanks!

**Review by WiNloSt — APPROVED** [review]

**Pull request merged**